src/snowcli/cli/streamlit/commands.py

```diff
@@ -144,3 +148,15 @@ def get_url(
     if open_:
         typer.launch(url)
     return MessageResult(url)
+
+
+def _override_connection_from_project_definition_if_no_flag(streamlit, options) -> None:
+    if streamlit.get("database") and not options.get("database"):
```

Comment:
\1 I'm not sure we should read manually from options and update them. IMO we should read the state from global context and update only global context.

My suggestion:
```python
database_from_project = streamlit.get("database")
if database_from_project and not cli_context_manager.connection_context.database:
   cli_context_manager.connection_context.set_database(database_from_project)
```

\2 I know that @sfc-gh-turbaszek wanted to avoid modifications of global context in commands implementations. Tomek wdyt about this case?

Reply:
1. When I get database from context I don't know if it was overwritten by flag or not

Reply:
database and schema were removed from this PR